node/impls.go

```diff
@@ -153,17 +153,26 @@ type agreementLedger struct {
 // EnsureBlock implements agreement.LedgerWriter.EnsureBlock.
 func (l agreementLedger) EnsureBlock(e bookkeeping.Block, c agreement.Certificate) {
 	l.Ledger.EnsureBlock(&e, c)
+	// let the network know that we've made some progress.
+	l.n.OnNetworkAdvance()
```

Comment:
No. If the agreement can't work due to gossip network issue, we don't want the catchup service to become the source of the new blocks. Instead, we want to disconnect, and hopefully, it would help fixing the gossip network.

Reply:
If the OnNewBlock is called *only* as a result of the agreement protocol advancements ( all the three ), then yes, it would be a good think to do. ( but indirectly; The node.go should be node that bridge the gap )